app/storage/metadata_parser.py

```diff
@@ -1,103 +1,78 @@
-import uuid
-from datetime import datetime
-from structlog import get_logger
+import functools
 
-from sdc.crypto.exceptions import InvalidTokenException
+from typing import List, Dict
+from structlog import get_logger
+from marshmallow import Schema, fields, validate, EXCLUDE, pre_load
 
 logger = get_logger()
 
 
-def iso_8601_date_parser(iso_8601_string):
-    return datetime.strptime(iso_8601_string, '%Y-%m-%d')
-
-
-def uuid_4_parser(plain_string):
-    return str(uuid.UUID(plain_string))
-
-
-def boolean_parser(boolean_value):
-    if not isinstance(boolean_value, bool):
-        raise TypeError('Claim was not of type `bool`')
-    return boolean_value
-
-
-def string_parser(string_value):
-    if not isinstance(string_value, str) or string_value == '':
-        raise TypeError('Claim not a valid string value')
-    return string_value
+class UUIDString(fields.UUID):
+    def _deserialize(self, *args, **kwargs):  # pylint: disable=arguments-differ
+        return str(super()._deserialize(*args, **kwargs))
 
 
-def optional_string_parser(optional_string_value):
-    if not isinstance(optional_string_value, str):
-        raise TypeError('Claim not a valid/empty string')
-    return optional_string_value
+VALIDATORS = {
+    'date': functools.partial(fields.DateTime, format='%Y-%m-%d', required=True),
+    'uuid': functools.partial(UUIDString, required=True),
+    'boolean': functools.partial(fields.Boolean, required=True),
+    'string': functools.partial(fields.String, required=True),
+    'url': functools.partial(fields.Url, required=True),
+}
 
 
-def clean_leading_trailing_spaces(metadata):
-    for key, value in metadata.items():
-        if isinstance(value, str):
-            metadata[key] = value.strip()
+class StripWhitespaceMixin:
+    @pre_load
+    def strip_whitespace(self, items):  # pylint: disable=no-self-use
+        for key, value in items.items():
+            if isinstance(value, str):
+                items[key] = value.strip()
+        return items
+
+
+class RunnerMetadataSchema(Schema, StripWhitespaceMixin):
+    jti = VALIDATORS['uuid'](validate=validate.Length(min=1))
```

Comment:
uuid validator with a min length of 1?

Reply:
What happens with a uuid with an empty string?

Reply:
Tested that empty strings are not valid UUIDs